deptofdefense/move.mil
# Add Entitlements pages

## Checklist

I have…


## Summary of Changes

This pull request adds the Entitlements page to the Rails app. On this page, service members can determine their moving allowances by searching based on their rank, dependency status, and move type. The search results display entitlements based on user input. Additionally, special item information is presented below search results.

From an implementation standpoint, this page renders thusly:

1. The server returns the page's markup, which includes a list of links to individual entitlement pages (e.g. http://localhost:3000/entitlements/e-1).
1. JavaScript is then executed to hide this list and display the search form.
1. A user entering a search and pressing the "View Results" button will initiate an AJAX request that, when successful, will return a block of markup.
1. That block of markup is injected into the search results container and the default "Information Needed" info box is hidden.
1. Subsequent user searches will replace the search results container's content.

## Testing

To verify the changes proposed in this pull request…

1. clone this repo,
1. `git checkout entitlements`,
1. set up development dependencies according to `CONTRIBUTING.md`,
1. `bin/rails server`,
1. load up http://localhost:3000/entitlements in your Web browser of choice, and
1. perform an entitlements search.

## Screenshots

<table>
<tr>
<td valign="top">
<h3>Without Search Results</h3>
<img src="https://user-images.githubusercontent.com/27780860/29620739-bdafa526-87ec-11e7-9522-071231370df7.png" alt="no search">
</td>
<td valign="top">
<h3>With Search Results</h3>
<img src="https://user-images.githubusercontent.com/27780860/29620745-c837fdfe-87ec-11e7-9f8d-379e1fcdd040.png" alt="search">
</td>
</tr>
</table>
